core/src/main/java/feign/ReflectiveFeign.java

```diff
@@ -201,6 +204,11 @@ public RequestTemplate create(Object[] argv) {
 
     protected RequestTemplate resolve(Object[] argv, RequestTemplate mutable,
                                       Map<String, Object> variables) {
+      if ("POST".equals(metadata.template().method()) && argv == null) {
+        // write an empty string for the BODY to conform with okhttp 2.4.0+
+        // http://johnfeng.github.io/blog/2015/06/30/okhttp-updates-post-wouldnt-be-allowed-to-have-null-body/
+        encoder.encode("", String.class, mutable);
```

Comment:
What other clients? You mean ones that don't use ReflectiveFeign? We only use the JAXRSContract so that's the targeted fix that I made here. I didn't test out other contracts.